crates/rune/src/indexing/index.rs

```diff
@@ -285,6 +289,46 @@ impl<'a> Indexer<'a> {
         }))
     }
 
+    /// Expand a macro returning the current file
+    fn expand_file_macro(&mut self, ast: &mut ast::MacroCall) -> Result<BuiltInMacro, ParseError> {
+        let id = self.storage.insert_str(self.source.name());
+        let source = ast::StrSource::Synthetic(id);
+        let node = ast::LitStr {
+            token: ast::Token {
+                kind: ast::Kind::Str(ast::StrSource::Synthetic(id)),
+                span: ast.span(),
+            },
+            source,
+        };
+        Ok(BuiltInMacro::File(BuiltInFile {
+            span: ast.span(),
+            value: node,
+        }))
+    }
+
+    /// Expand a macro returning the current line for where the macro invocation begins
+    fn expand_line_macro(&mut self, ast: &mut ast::MacroCall) -> Result<BuiltInMacro, ParseError> {
+        let (l, _) = self
+            .source
+            .position_to_utf16cu_line_char(ast.open.span.start.into_usize())
```

Comment:
hah, not that it matters (line count should be the same), but it would be nice with an utf8 version of this as well.

Reply:
How'd that work - I can't see any information about the actual encoding of the incoming source? This should probably use the unicode variant though, not utf16.

Reply:
`utf16cu` stands for UTF-16 code units, the position function does some work to re-encode each character to utf-16 in order to find the column. An equivalent version for UTF-8 would simply rely on the emitted chars instead.

See: https://github.com/rune-rs/rune/blob/master/crates/runestick/src/source.rs#L135

Not a blocker though, since the line will be the same.